File: pset2/readability/readability.c
#include<stdio.h>
#include<string.h>
#include<ctype.h>
#include<cs50.h>
#include<math.h>

void readability(string text);
int countLettersOfAnString(string text);
int countWordsOfAnString(string text);
int countSentencesOfAnString(string text);
int calculateIndex(int letters, int words, int sentences);
void printGrade(int index);

int main(void)
{
    readability(get_string("Text:"));
}

void readability(string text)
{
    int letters = countLettersOfAnString(text);
    int words = countWordsOfAnString(text);
    int sentences = countSentencesOfAnString(text);
    int index = calculateIndex(letters, words, sentences);
    
    printGrade(index);

}

int countLettersOfAnString(string text)
{
    int letters = 0;
    
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    
    return letters;
}

int countWordsOfAnString(string text)
{
    int spaces = 0;
    
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            spaces++;
        }
    }
    
    return spaces + 1;
}

int countSentencesOfAnString(string text)
{
    int sentences = 0;
    
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?') 
        {
            sentences++;
        }
    }
    
    return sentences;
}

int calculateIndex(int letters, int words, int sentences)
{
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    
    float idx = (0.0588 * L - 0.296 * S - 15.8);

    return (idx - ((int) idx)) > 0.5 ? ceil(idx) : idx;
}

void printGrade(int index)
{
    if (index < 0)
    {
        printf("Before Grade 1\n");
    } 
    else if (index > 16)
    {
        printf("Grade 16+\n");
    } 
    else
    {
        printf("Grade %d\n", index);
    }
}

File: pset6/readability/readability.py
from cs50 import *


def main():
    readability(get_string("Text:"))


def readability(text):
    letters = countLettersOfAnString(text)
    words = countWordsOfAnString(text)
    sentences = countSentencesOfAnString(text)
    index = calculateIndex(letters, words, sentences)

    printGrade(index)


def countLettersOfAnString(text):
    letters = 0

    for i in range(len(text)):
        if (text[i].isalpha()):
            letters += 1

    return letters


def countWordsOfAnString(text):
    spaces = 0

    for i in range(len(text)):
        if (text[i] == ' '):
            spaces += 1

    return spaces + 1


def countSentencesOfAnString(text):
    sentences = 0

    for i in range(len(text)):
        if (text[i] == '.' or text[i] == '!' or text[i] == '?'):
            sentences += 1

    return sentences


def calculateIndex(letters, words, sentences):
    L = (letters / words) * 100
    S = (sentences / words) * 100

    idx = (0.0588 * L - 0.296 * S - 15.8)

    return round(idx)


def printGrade(index):
    if (index < 0):
        print("Before Grade 1")

    elif (index > 16):
        print("Grade 16+")

    else:
        print(f"Grade {index}")


main()